src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/hmi/rc_set_interior_vehicle_data_request.cc

```diff
@@ -0,0 +1,29 @@
+#include "rc_rpc_plugin/commands/hmi/rc_set_interior_vehicle_data_request.h"
+#include "utils/macro.h"
+
+namespace rc_rpc_plugin {
+namespace commands {
+
+RCSetInteriorVehicleDataRequest::RCSetInteriorVehicleDataRequest(
+    const app_mngr::commands::MessageSharedPtr& message,
+    app_mngr::ApplicationManager& application_manager,
+    app_mngr::rpc_service::RPCService& rpc_service,
+    app_mngr::HMICapabilities& hmi_capabilities,
+    policy::PolicyHandlerInterface& policy_handle,
+    ResourceAllocationManager& resource_allocation_manager)
+    : application_manager::commands::RequestToHMI(message,
+                                                  application_manager,
+                                                  rpc_service,
+                                                  hmi_capabilities,
+                                                  policy_handle) {
+  UNUSED(resource_allocation_manager);
+}
+RCSetInteriorVehicleDataRequest::~RCSetInteriorVehicleDataRequest() {}
+
+void RCSetInteriorVehicleDataRequest::Run() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  SendRequest();
```

Comment:
how the request timeout is managed?
in general we have a default 10 seconds timeout period, if a request cannot be finished within that allowed time period, sdl gives a generic error to the request.

for the special case that sdl send GetInteriorVehicleDataConsent  to HMI ask for driver's permission, sdl need to wait another 10 seconds for GetInteriorVehicleDataConsent  and thus 20 seconds for SetInteriorVehicleDataRequest. Where are those logic implemented?